quattor/types/metadata.pan

```diff
@@ -14,7 +14,35 @@ type structure_template = {
     "branch" : structure_branch
 };
 
+type structure_os_version = {
+    @{Descriptive form of version, e.g. 7x}
+    "name" : string
+    @{Major version number component, e.g. 7}
+    "major" ? long(0..)
+    @{Minor version number component}
+    "minor" ? long(0..)
+};
+
+type structure_os_distribution = {
+    @{Distribution name, e.g. sl}
+    "name" : string with match(SELF, '^((rh)?el|centos|sl)$')
+    @{Full descriptive name, e.g. Scientific Linux}
+    "description" ? string
+    @{Family that this distribution belongs to e.g. The family for "sl" would be "el"}
+    "family" ? string with match(SELF, '^el$')
+};
+
+type structure_os = {
+    @{OS distribution details}
+    "distribution" : structure_os_distribution
+    @{OS version details}
+    "version" : structure_os_version
+    @{Architecture of OS (may be different to system architecture)}
+    "architecture" : string
+};
+
 type structure_metadata = {
     "template" : structure_template
     "features" ? string[]
+    "os" ? structure_os
```

Comment:
can we add some annotation to the aquilon schema to describe the metadata a bit better; incl the `structure_metadata` itself (or at least what we should and should add to `/metadata`.